sstables/index_reader.hh

```diff
@@ -318,11 +319,12 @@ public:
 
     index_consume_entry_context(const sstable& sst, reader_permit permit, IndexConsumer& consumer, trust_promoted_index trust_pi,
             input_stream<char>&& input, uint64_t start, uint64_t maxlen,
-            std::optional<column_values_fixed_lengths> ck_values_fixed_lengths, tracing::trace_state_ptr trace_state = {})
+            std::optional<column_values_fixed_lengths> ck_values_fixed_lengths, tracing::trace_state_ptr trace_state = {}, abort_source* abort = nullptr)
```

Comment:
What callers don't supply it?

It those are just tests, maybe it's better to make it mandatory and have the tests supply a dummy abort_source. Default parameters are an invitation to missing parameters.

Reply:
Now the index_consume_entry_context gets the abort_source directly from the sstable/sstable_manager. Updated the tests as well.